environments/contrail/install_vrouter_kmod.yaml

```diff
@@ -98,8 +103,17 @@ resources:
             protect=1
             metadata_expire=30
             EOF
-              yum install -y contrail-tripleo-puppet puppet-contrail
-              rsync -a /usr/share/contrail-tripleo-puppet/ /usr/share/openstack-puppet/modules/tripleo/
+            fi
+            if [[ ${contrail_repo_new} ]]; then
```

Comment:
Why do you delete yum install ?

Reply:
Am I correct that all control plane nodes (both old and new) will be populated with both old and new repos? Is it ok?

Reply:
Deleting yum install: There are package upgrade issues (bugs probably). So working around by not installing the old packages on the new nodes at all. The user can use artifacts to transfer puppet modules to the overcloud nodes. The idea is to install the new packages *only* on the new contrail-controllers.

Only the new control plane nodes will be populated with both repos. Should be okay.

Reply:
but deleting of 'yum install' breaks current clean installation case that doesn't require swift artefacts for now.

Reply:
Yes, that's true. I'd been using artifacts in the 5b5 cluster to deploy overcloud. This method will always require uploading puppet modules via artifacts.